### PR TITLE
New package: uxn-1.0

### PR DESCRIPTION
**Pull request by Eloitor**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-musl)

### CODE REVIEW

**Comment by Eloitor** [comment]
I don't know how to fix cross compilation...

for aarch64 I get:
```
cc1: error: bad value 'armv8-a' for '-march=' switch
cc1: note: valid arguments to '-march=' switch are: nocona core2 nehalem corei7 westmere sandybridge corei7-avx ivybridge core-avx-i haswell core-avx2 broadwell skylake skylake-avx512 cannonlake icelake-client rocketlake icelake-server cascadelake tigerlake cooperlake sapphirerapids alderlake bonnell atom silvermont slm goldmont goldmont-plus tremont knl knm x86-64 x86-64-v2 x86-64-v3 x86-64-v4 eden-x2 nano nano-1000 nano-2000 nano-3000 nano-x2 eden-x4 nano-x4 k8 k8-sse3 opteron opteron-sse3 athlon64 athlon64-sse3 athlon-fx amdfam10 barcelona bdver1 bdver2 bdver3 bdver4 znver1 znver2 znver3 btver1 btver2 native
```

File: srcpkgs/uxn/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+makedepends="SDL2-devel"
+short_desc="Assembler and emulator for the Uxn stack-machine"
+maintainer="Eloi Torrents <eloitor@duck.com>"
+license="MIT"
+homepage="https://100r.co/site/uxn.html"
+distfiles="https://git.sr.ht/~rabbits/uxn/archive/${version}.tar.gz"
+checksum=29059ee288474e48fae4b9e755bbc8e1b392a9c7ac90449d7a6599385c145460
+
+do_build() {
+	CC="$BUILD_CC" CFLAGS="$XBPS_CROSS_CFLAGS" ./build.sh
```
Comment (Johnnynator) [inline srcpkgs/uxn/template:14]: You don't want to mess with CC/CFLAGS, they are already set correctly for cross compiling.
You want to add `--no-run`, so it does not attempt to run the just compiled binaries.

The other problem is, the `build.sh` adds `-g0` to `CFLAGS`, which we don't want. The binaries will get stripped by `void-packages`

File: srcpkgs/uxn/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+do_install() {
+	vbin bin/uxnasm
+	vbin bin/uxncli
+	vbin bin/uxnemu
+}
+
+post_install() {
+	vlicense LICENSE
```
Comment (Johnnynator) [inline srcpkgs/uxn/template:24]: put it into `do_install`.

File: srcpkgs/uxn/template (hunk 1 above)
Comment (Eloitor) [reply srcpkgs/uxn/template:14]: Thank you. Is it a good idea to just put the 3 lines needed for compilation in `do_build` and ignore the `build.sh` file?

Comment (Johnnynator) [reply srcpkgs/uxn/template]: Imo just patching it is better.
Vendoring parts of the script will add an additional burden of keeping it up to date.

**Pull request closed without merging**